src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/GraphNodes/DependenciesGraphProvider.cs

```diff
@@ -340,9 +339,6 @@ private static GraphNodeId GetTopLevelGraphNodeId(string projectPath, string mod
                 GraphNodeId.GetPartial(CodeGraphNodeIdName.File, filePath));
         }
 
-        private static string GetIconStringName(ImageMoniker icon)
-        {
-            return $"{icon.Guid.ToString()};{icon.Id}";
-        }
+        private static string GetIconStringName(ImageMoniker icon) => $"{icon.Guid:D};{icon.Id}";
```

Comment:
Since D is the default for Guid.ToString() it might be better to remove the formatter specifier to be sure the compiler optimizes this to string concatenation

Reply:
Looks like the [compiler emits](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQFEAeAHJzkCWA9gHYBqUCBUwANigDQwgG0A+AAgAwAEHAjAG4AsACgxHAMx8ATDwDCYgN5ieavtI4AWHgFkAFAHE4BACY8A5idMBKVepWj1zngDdKPZPx4BeHgBIAImMzHyUrMwBfQJEnFzV3BE85PyCQ0zCI0xAAEWjY50ixSKA) a call to `System.String::Format` with the `D` specifier in the format string.

Reply:
Looks like the optimization I was thinking of only applies to strings, so you have to call .ToString() on the guid to get it, and use the C# 8 compiler. Without those things the `:D` makes no difference.

https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQGMoBssQaiADwAcllkBLAewDsAaGECrAHwAEAGAAjYEYBuALAAoEWwDMPAExcAwiIDeIrip6S2AFi4BZABQBxOBQAmXAOZHjASmWqlw1Y64A3KAi7JeXALxcAJABEhibeChYmAL4BQg5OKq7uyDK+gcHGoeHGIAAiUTFx8W4ekilBlhmWAHQAKlQAyjAIFDRmulZ5tioRIhFAA==